### PR TITLE
Various issues and additions

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Various additions and issues found during writing examples:

- Free function for sorting events in a dataset.
- Operations with subsets now works (`dataset.subset['name'] += ...`). Note the change from `subset` to a property that was necessary to do this.
- Missing `__setitem__` for slices of datasets.
- Added helper to nicely format 1D datasets in Jupyter as a table.
  ```python
   import dataset as ds
   ds.table(mydataset)
   ```
- Add missing `py::keep_alive` to avoid segmentation faults when iterating "temporaries".

### CODE REVIEW

**Comment by igudich** [comment]
May be `dataset.variables['name'] += ...` instead of `subset`?

File: exports/dataset.cpp [hunk 1]
```diff
@@ -492,6 +492,18 @@ using as_VariableView =
                         std::string, boost::container::small_vector<double, 8>,
                         Dataset, Eigen::Vector3d>;
 
+class SubsetHelper {
```
Comment (OwenArnold) [inline exports/dataset.cpp]: Maybe SubsetFacade or SubsetWrapper would be better as a name

File: python/table.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+def table(dataset):
```
Comment (OwenArnold) [inline python/table.py]: Did you consider not hand-rolling this as a string? i.e use https://pypi.org/project/htmlBuilder/, http://www.yattag.org/ maybe others?

Comment (SimonHeybrock) [reply python/table.py]: Yes I did, but this was faster as a quick hack for now. It should probably be extended anyway to list 0-D variables separately, and deal with bin-edges, etc.. I will add this to the TODO list.

File: exports/dataset.cpp (hunk 1 above)
Comment (SimonHeybrock) [reply exports/dataset.cpp]: Aren't "facade" as well as "Wrapper" misleading? It is not a facade for a subset, and not a wrapper for a subset. It is a helper for creating subsets. Anyway, it is just used locally, so unless this is used more widely...

**Comment by SimonHeybrock** [comment]
@igudich 

> May be `dataset.variables['name'] += ...` instead of `subset`?

I don't think so, the "subset" is very important. Consider:

File: exports/dataset.cpp (hunk 1 above)
Comment (OwenArnold) [reply exports/dataset.cpp]: Ok, i'm not particularly concerned about this. It was this https://github.com/mantidproject/dataset/pull/77/files#diff-ed677a2708c9afda3b34f34bde8de2b0R814 that prompted me to raise the question.

**Pull request merged**